### PR TITLE
Database 1.6.0 Reference (M4)

### PR DESCRIPTION
**Pull request by george-hoffer**

### CODE REVIEW

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 1]
```diff
@@ -294,13 +283,13 @@ Allows executing one insert statement various times using different parameter bi
...
+** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
 | Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
 | Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
 | SQL Query Text a| String |  The text of the SQL query to execute |  | x
-| Parameter Types a| Array of <<ParameterType>> |  Allows you to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 
+| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter not present in the input values. |  | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 2]
```diff
@@ -348,13 +337,13 @@ Allows executing one update statement various times using different parameter bi
...
+** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
+| Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
 | Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
 | SQL Query Text a| String |  The text of the SQL query to execute |  | x
-| Parameter Types a| Array of <<ParameterType>> |  Allows you to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter that's not present in the input values. |  | 
+| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter not present in the input values. |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| Parameter Types a| Array of <<ParameterType>> |  Allows you to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter not present in the input values. |  | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 3]
```diff
@@ -402,14 +390,14 @@ Deletes data in a database.
...
-| SQL Query Text a| String |  The text of the SQL query to execute. |  | x
-| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter that is not present in the input values. |  | 
-| Input Parameters a| Object |  A Map in which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: where id = :myParamName)).  The map's values will contain the actual assignation for each parameter. |  | 
+| SQL Query Text a| String |  The text of the SQL query to execute |  | x
+| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]: The word "even" is unnecessary
Add a period to the end of the sentence

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 4]
```diff
@@ -402,14 +390,14 @@ Deletes data in a database.
...
-| Input Parameters a| Object |  A Map in which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: where id = :myParamName)).  The map's values will contain the actual assignation for each parameter. |  | 
+| SQL Query Text a| String |  The text of the SQL query to execute |  | x
+| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 
+| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]: "semicolon" is wrong and I fixed it last time I updated this reference (which was recently). It's a colon, not a semicolon. This was incorrect in multiple places. I did a S&R. Please change back to the correct colon.

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 5]
```diff
@@ -564,17 +549,17 @@ Inserts data into a Database
...
 | SQL Query Text a| String |  The text of the SQL query to execute |  | x
 | Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 
-| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: where id = :myParamName)).  The map's values will contain the actual assignation for each parameter. |  | 
+| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contains the actual assignment for each parameter. |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (for example, where `ID = :myParamName`).  The map's values contains the actual assignment for each parameter. |  | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 6]
```diff
@@ -625,14 +609,14 @@ Selects data from a database. Streaming is automatically applied to avoid preemp
...
 | SQL Query Text a| String |  The text of the SQL query to execute |  | x
 | Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 
-| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: where id = :myParamName)).  The map's values will contain the actual assignation for each parameter. |  | 
+| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 7]
```diff
@@ -679,19 +662,20 @@ Invokes a Stored Procedure on the database.  When the stored procedure returns o
...
-| Input - Output Parameters a| Object |  A Map which keys are the name of a parameter to be set on the JDBC prepared statement which is both input and output.  Each parameter should be referenced in the SQL text using a colon prefix (E.g: where id = :myParamName)).  The map's values will contain the actual assignation for each parameter. |  | 
-| Output Parameters a| Array of <<OutputParameter>> |  A list of output parameters to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: call multiply(:value, :result)) |  | 
+| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 8]
```diff
@@ -679,19 +662,20 @@ Invokes a Stored Procedure on the database.  When the stored procedure returns o
...
-| Output Parameters a| Array of <<OutputParameter>> |  A list of output parameters to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: call multiply(:value, :result)) |  | 
+| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  | 
+| Input - Output Parameters a| Object |  A Map which keys are the name of a parameter to be set on the JDBC prepared statement which is both input and output.  Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values will contain the actual assignment for each parameter. |  | 
+| Output Parameters a| Array of <<OutputParameter>> |  A list of output parameters to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, call `multiply(:value, :result)`) |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| Output Parameters a| Array of <<OutputParameter>> |  A list of output parameters to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (for example, call `multiply(:value, :result)`) |  | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 9]
```diff
@@ -739,17 +723,17 @@ Updates data in a database.
...
 | SQL Query Text a| String |  The text of the SQL query to execute |  | x
 | Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 
-| Input Parameters a| Object |  A map in which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: where id = :myParamName)).  The map's values will contain the actual assignation for each parameter. |  | 
+| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (E.g: where ID = :myParamName)).  The map's values will contain the actual assignment for each parameter. |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: where ID = :myParamName)).  The map's values will contain the actual assignment for each parameter. |  | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 10]
```diff
@@ -764,38 +748,40 @@ Updates data in a database.
...
 ==== Parameters
 [%header,cols="20s,20a,35a,20a,5a"]
 |===
 | Name | Type | Description | Default Value | Required
 | Configuration | String | The name of the configuration to use. | | x
 | Table a| String |  The name of the table to select from |  | x
-| Watermark Column a| String |  The name of the column to use for watermark. Values taken from this column will be used to filter the contents of the next poll, so that only rows with a greater watermark value are processed. |  | 
-| Id Column a| String |  The name of the column to consider as row ID. If provided, this component will make sure that the same row is not processed twice by concurrent polls. |  | 
+| Watermark Column a| String |  The name of the column to use for watermark. Values taken from this column are used to filter the contents of the next poll, so that only rows with a greater watermark value are processed. |  | 
+| Id Column a| String |  The name of the column to consider as row ID. If provided, this component makes sure that the same row is not processed twice by concurrent polls. |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| Id Column a| String |  The name of the column to consider as row ID. If provided, this component ensures that the same row is not processed twice by concurrent polls. |  | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 11]
```diff
@@ -764,38 +748,40 @@ Updates data in a database.
...
+| Id Column a| String |  The name of the column to consider as row ID. If provided, this component makes sure that the same row is not processed twice by concurrent polls. |  | 
 | Transactional Action a| Enumeration, one of:
 
 ** ALWAYS_BEGIN
 ** NONE |  The type of beginning action that sources can take regarding transactions. |  NONE | 
 | Transaction Type a| Enumeration, one of:
 
 ** LOCAL
-** XA |  The type of transaction to create. Availability depends on the runtime version. |  LOCAL | 
+** XA |  The type of transaction to create. Availability depends on Mule version. |  LOCAL |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
** XA |  The type of transaction to create. Availability depends on the Mule version. |  LOCAL | 
```

File: modules/ROOT/pages/db/database-documentation.adoc [hunk 12]
```diff
@@ -1007,7 +998,7 @@ Selects from a table at a regular interval and generates one message per each ob
 [%header,cols="20s,25a,30a,15a,10a"]
 |===
 | Field | Type | Description | Default Value | Required
-| In Memory Objects a| Number | The maximum amount of instances that will be kept in memory. If more than that is required, then it will start to buffer the content on disk. |  | 
+| In Memory Objects a| Number | The maximum amount of instances to keep in memory. If more than that is required, then it will start to buffer the content on disk. |  |
```
Comment (valkyrie69) [inline modules/ROOT/pages/db/database-documentation.adoc]:
```suggestion
| In Memory Objects a| Number | The maximum number of instances to keep in memory. If more than that is required, content on the disk is buffered. |  | 
```

**Review by valkyrie69 — COMMENTED** [review]
Mostly minor. Only one that's important. It's colon, not semicolon.

File: modules/ROOT/pages/db/database-documentation.adoc (hunk 4 above)
Comment (george-hoffer) [reply modules/ROOT/pages/db/database-documentation.adoc]: Fixed throughout

**Review by valkyrie69 — APPROVED** [review]

**Comment by valkyrie69** [comment]
Updated reference for 1.6.2 so there is a more recent PR. (PR #554)

**Pull request closed without merging**